### DISCUSSION TITLE
Sdk - readMany - filter typing

### DISCUSSION THREAD

**Discussion by gkielwasser** [directus/directus#10132] [Archive]
Hi,

Since the sdk 9.1.2 I have the filter typing that is broken for the following use case.

Here is how I was getting a typed products results from the readMany api:
```
const { data: products } = await directus
      .items<string, Products>('products')
      .readMany({
        filter: {
          my_field: {
            _in: ["id1","id2"],
          },
        },
      })
```

Now the compiler complains that "_in" should be a string instead of a string[]. It seems wrong.
```
TS2322: Type 'string[]' is not assignable to type 'string'.
```

Note: In the case I don't type items, the filter is of type any and obviously I don't have any errors.
```
[...]
.items('products')
[...]
```

Any idea why it has stopped working?

**Comment by joselcvarela** [comment]
Hello @gkielwasser 
We have recently improved the typescript definitions for filter to support `_and` and `_or`. It seems we forgot the array definitions.
https://github.com/directus/directus/pull/9965

But I already have pushed a fix:
https://github.com/directus/directus/pull/10133

**Reply by gkielwasser** [reply]
Hi @joselcvarela 
I see, thanks for the quick fix :)
